packages/client/src/generation/TSClient/common.ts

```diff
@@ -416,6 +416,16 @@ type _Record<K extends keyof any, T> = {
   [P in K]: T;
 };
 
+// cause typescript not to expand types and preserve names
+type NoExpand<T> = T extends unknown ? T : never;
+
+// this type assumes the passed object is entirely optional
+type AtLeast<O extends object, K extends string> = NoExpand<
```

Comment:
What would you think about pulling this type into separate file and writing a couple of test for it using `expectTypeOf` ?